src/syrupy/assertion.py

```diff
@@ -377,11 +377,7 @@ def _recall_data(
     ) -> Tuple[Optional["SerializableData"], bool]:
         try:
             return (
-                self.extension.read_snapshot(
-                    test_location=self.test_location,
-                    index=index,
-                    session_id=str(id(self.session)),
```

Comment:
I note that #606 added this `session_id` parameter, which I imagine was meant to help with circumstances like this, but it seems like it ends up ignored/unused in practice?

- Amber doesn't use `cache_key`: https://github.com/tophat/syrupy/blob/ef8189c68460593fead9c484405b755e272c8cca/src/syrupy/extensions/amber/__init__.py#L57-L66
- Single-file doesn't use `session_id`: https://github.com/tophat/syrupy/blob/ef8189c68460593fead9c484405b755e272c8cca/src/syrupy/extensions/single_file.py#L92-L103

And, in practice, I think it may be quite hard to use, since the relevant cached data is held in memory in the session itself (i.e. the extension `_read_snapshot_data_from_location` method calls don't really have access to it at all).

Reply:
The session_id is used as the cache_key argument in Amber's `__cacheable_read_snapshot` method. Although it's not used in the function, it's used by the lru_cache decorator (which cases based on the kwargs I believe). So it essentially invalidates the cache